lib/utils/renderTree.ts

```diff
@@ -6,12 +6,16 @@ import {BlueshellState} from '../nodes/BlueshellState';
 import {serializeArchyTree} from './archyTree';
 import {serializeDotTree} from './dotTree';
 
-export function toString<S extends BlueshellState, E>(tree: Base<S, E>, state?: S): string {
-	return serializeArchyTree(tree, state);
+export function toString<S extends BlueshellState, E>(
+	tree: Base<S, E>, state?: S, contextDepth = Number.MAX_SAFE_INTEGER
```

Comment:
what about using `contextDepth?: number` so that the default only needs to be set in one place?

Reply:
I couldn't remember how JS handles (3, undefined) for (number, k = 10) -- I was worried explicit undefined would override the default an never went back to check.

Reply:
Forgot another thing: this forces flipping parameter order, because optional params (state, which is optional all the way down) can't precede required params. Went through the code and fixed this up -- buildArchyTree is a bit weird now, but it's not exported, so... `¯\_(ツ)_/¯`